### PR TITLE
[MM-48220] Implement job service for Calls

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

I am not proud of this code but in the spirit of "something is better than nothing" I am happy to give it a go. There's much to polish and review but this is a base.

#### Related PR

https://github.com/mattermost/mattermost-plugin-calls/pull/247

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-48220

### CODE REVIEW

File: service/jobs_api.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		return
+	}
+
+	if err := cfg.IsValid(); err != nil {
+		data.err = err.Error()
+		data.code = http.StatusBadRequest
+		return
+	}
+
+	if s.cfg.Jobs.APIType != JobAPITypeDocker || cfg.Type != JobTypeRecording {
```
Comment (cpoile) [inline service/jobs_api.go:41]: This gets checked in either of the `isValid` methods, right?

File: service/jobs_api.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+		data.err = "not implemented"
+		data.code = http.StatusNotImplemented
+		return
+	}
+
+	job, err := s.jobService.CreateRecordingJobDocker(cfg, func(job Job) error {
```
Comment (cpoile) [inline service/jobs_api.go:47]: But this should be conditional on the job APIType

File: service/jobs_service.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+	ctx, cancel := context.WithTimeout(context.Background(), dockerRequestTimeout)
+	defer cancel()
+	cli, err := docker.NewClientWithOpts(docker.FromEnv, docker.WithAPIVersionNegotiation())
+	if err != nil {
+		return Job{}, fmt.Errorf("failed to create docker client: %w", err)
+	}
+	defer cli.Close()
+
+	cnts, err := cli.ContainerList(ctx, types.ContainerListOptions{})
```
Comment (cpoile) [inline service/jobs_service.go:60]: `cnts`? 😂

**Review by cpoile — APPROVED** [review]
This is cool! I can't really review  the correctness of the meaty bit (jobs_service), but that's nice we can use docker like this.

File: service/jobs_api.go (hunk 1 above)
Comment (streamer45) [reply service/jobs_api.go:41]: Yes, the job type check is redundant at the moment but as soon as we have more we'll likely need to switch and call the appropriate method.

File: service/jobs_api.go (hunk 2 above)
Comment (streamer45) [reply service/jobs_api.go:47]: Correct

File: service/jobs_service.go (hunk 3 above)
Comment (streamer45) [reply service/jobs_service.go:60]: This is not candyland javascript :warning:

**Comment by streamer45** [comment]
I added some comments as I realize I did a poor job at explaining any of this. Will try to add more as I polish this next week.

**Comment by streamer45** [comment]
Merging this now. I'll prepare new PRs for any additional polish.

**Pull request merged**